client/src/QnA/QnA.jsx

```diff
@@ -2,50 +2,77 @@ import React from 'react';
 import axios from 'axios';
 import QnAList from './QnAList.jsx';
 import AddQuestion from './AddQuestion.jsx';
+import SearchQuestions from './SearchQuestions.jsx';
 
 const { useState, useEffect } = React;
 
 function QnA({ id, product }) {
   const [quests, setQuests] = useState([]);
+  const [filtQuests, setFiltQuests] = useState(quests);
   const [questModal, setQuestModal] = useState(false);
 
   const grabQuestions = () => {
+    // console.log('grabQuestions is invoked');
     const config = { params: { product_id: id, page: 1, count: 99999 } };
     axios.get('/qa/questions', config)
       .then((response) => {
         setQuests(response.data.results);
       })
       .catch((error) => {
-        console.log(error);
+        console.log('Error grabbing questions', error);
       });
   };
 
   const questModalClicker = () => {
     setQuestModal(!questModal);
   };
 
+  const filterQuestion = (input) => {
+    const filtered = quests.filter((quest) => quest.question_body.toLowerCase()
+      .includes(input.toLowerCase()));
+    // Promise.all(filtered)
+    //   .then((result) => {
+    //     setFiltQuests(result);
+    //   })
+    //   .catch((err) => {
+    //     console.log(err);
+    //   });
+    setFiltQuests(filtered);
+  };
+
   useEffect(() => {
     grabQuestions();
   }, []);
-
+  // Need to render change in selection from related products
   useEffect(() => {
     grabQuestions();
   }, [id]);
```

Comment:
In your QnA component, you have two useEffects - one with an empty array dependency and another that fires whenever id changes. I'm not entirely sure, but I think you can omit the empty dependency useEffect because the one that is dependent on id changes should still fire at the first render. Since they invoke the same helper function, I don't think this would break anything.